lib/src/ui/paged_sliver_builder.dart

```diff
@@ -141,70 +142,94 @@ class _PagedSliverBuilderState<PageKeyType, ItemType>
-                    _newPageProgressIndicatorBuilder,
-                  );
-                case PagingStatus.completed:
-                  return widget.completedListingBuilder(
-                    context,
-                    _buildListItemWidget,
-                    _itemCount,
-                    _noMoreItemsIndicatorBuilder,
-                  );
-                case PagingStatus.loadingFirstPage:
-                  return _FirstPageStatusIndicatorBuilder(
-                    builder: _firstPageProgressIndicatorBuilder,
-                    shrinkWrap: _shrinkWrapFirstPageIndicators,
-                  );
-                case PagingStatus.subsequentPageError:
-                  return widget.errorListingBuilder(
-                    context,
-                    _buildListItemWidget,
-                    _itemCount,
-                    (context) => _newPageErrorIndicatorBuilder(context),
-                  );
-                case PagingStatus.noItemsFound:
-                  return _FirstPageStatusIndicatorBuilder(
-                    builder: _noItemsFoundIndicatorBuilder,
-                    shrinkWrap: _shrinkWrapFirstPageIndicators,
-                  );
-                default:
-                  return _FirstPageStatusIndicatorBuilder(
-                    builder: _firstPageErrorIndicatorBuilder,
-                    shrinkWrap: _shrinkWrapFirstPageIndicators,
+              break;
+            case PagingStatus.completed:
+              child = widget.completedListingBuilder(
+                context,
+                _buildListItemWidget,
+                _itemCount,
+                _noMoreItemsIndicatorBuilder,
+              );
+              break;
+            case PagingStatus.loadingFirstPage:
+              child = _FirstPageStatusIndicatorBuilder(
+                builder: _firstPageProgressIndicatorBuilder,
+                shrinkWrap: _shrinkWrapFirstPageIndicators,
+              );
+              break;
+            case PagingStatus.subsequentPageError:
+              child = widget.errorListingBuilder(
+                context,
+                _buildListItemWidget,
+                _itemCount,
+                (context) => _newPageErrorIndicatorBuilder(context),
+              );
+              break;
+            case PagingStatus.noItemsFound:
+              child = _FirstPageStatusIndicatorBuilder(
+                builder: _noItemsFoundIndicatorBuilder,
+                shrinkWrap: _shrinkWrapFirstPageIndicators,
+              );
+              break;
+            default:
+              child = _FirstPageStatusIndicatorBuilder(
+                builder: _firstPageErrorIndicatorBuilder,
+                shrinkWrap: _shrinkWrapFirstPageIndicators,
+              );
+          }
+
+          // The SliverPadding is used to avoid changing the topmost item inside
+          // a CustomScrollView.
+          // https://github.com/flutter/flutter/issues/55170
+          child = SliverPadding(
```

Comment:
I don't think we need `SliverPadding` when animating transitions. I was thinking of two ways of solving this:
1. Only use `SliverPadding` if we're not animating transitions.
2. Always use `SliverAnimatedSwitcher`, but give it a 0 seconds duration if `animateTransitions` is `false`.

I think I like the second approach better. What are your thoughts?